### PR TITLE
CI: Install geopandas using pip in the "GMT Dev Tests" workflow

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Xref https://github.com/GenericMappingTools/pygmt/pull/2125#discussion_r1059909189 and https://github.com/GenericMappingTools/pygmt/pull/1290#issuecomment-847495768

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: .github/workflows/ci_tests_dev.yaml [hunk 1]
```diff
@@ -133,14 +132,6 @@ jobs:
 
       # Pull baseline image data from dvc remote (DAGsHub)
       - name: Pull baseline image data from dvc remote
```
Comment (seisman) [inline .github/workflows/ci_tests_dev.yaml:134]: Lines 134-139 and lines 141-146 are duplicated.

Comment (weiji14) [reply .github/workflows/ci_tests_dev.yaml:134]: Yeah, looks like we forgot to remove these lines in #2045.

**Review by weiji14 — APPROVED** [review]
Cool, thanks @seisman. This should allow us to catch any GDAL version conflicts between GMT and geopandas/fiona, and solve them proactively on conda-forge ahead of time!

**Pull request merged**